dvc/stage/__init__.py

```diff
@@ -463,15 +463,21 @@ def compute_md5(self):
         logger.debug("Computed %s md5: '%s'", self, m)
         return m
 
-    def save(self, allow_missing: bool = False, merge_versioned: bool = False):
+    def save(
+        self,
+        allow_missing: bool = False,
+        merge_versioned: bool = False,
+        run_cache: bool = True,
+    ):
         self.save_deps(allow_missing=allow_missing)
 
         self.save_outs(
             allow_missing=allow_missing, merge_versioned=merge_versioned
         )
         self.md5 = self.compute_md5()
 
-        self.repo.stage_cache.save(self)
+        if run_cache:
+            self.repo.stage_cache.save(self)
```

Comment:
Will this turn off saving to the run cache for `dvc repro --no-run-cache`? I think this is unclear from the help text:

>   --no-run-cache       Execute stage commands even if they have already been run with
                       the same command/dependencies/outputs/etc before.

The way I read it, this option should disable reading from the run cache but not writing to it.

Reply:
How you read it is how it works.

I can see how the options can be confusing. The changes here only affect `--no-commit` to make sure that there is no writing to the cache.

I am not sure about how to improve the flags, but the idea of the P.R. is to respect the description of `--no-commit` option:

> Do not store the outputs of this execution in the cache ([dvc.yaml](https://dvc.org/doc/user-guide/project-structure/dvcyaml-files) and [dvc.lock](https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#dvclock-file) are still created or updated); useful to avoid caching unnecessary data when exploring different data or stages.